contracts/libs/FixedPointInt256.sol

```diff
@@ -0,0 +1,101 @@
+/**
+ * SPDX-License-Identifier: UNLICENSED
+ */
+pragma solidity 0.6.10;
+
+import "../packages/oz/SignedSafeMath.sol";
+
+/**
+ *
+ */
+library FixedPointInt256 {
+    using SignedSafeMath for int256;
+
+    int256 private constant SCALING_FACTOR = 1e18;
+
+    /**
+     * @notice convert an unsigned integer to signed integer
+     * @param a uint to convert into a signed integer.
+     * @return the converted signed integer.
+     */
+    function fromUint(uint256 a) internal pure returns (int256) {
+        require(a < uint256(-1), "FixedPointInt256: out of int range");
+
+        return int256(a);
+    }
+
+    /**
+     * @notice convert a signed integer to unsigned integer
+     * @param a int to convert into an unsigned integer.
+     * @return the converted unsigned integer.
+     */
+    function fromInt(int256 a) internal pure returns (uint256) {
+        if (a < 0) {
+            return uint256(-a);
+        } else {
+            return uint256(a);
+        }
+    }
+
+    /**
+     * @notice return the sum of two signed integer
+     * @param a signed integer
+     * @param b signed integer
+     * @return sum of two signed integer
+     */
+    function add(int256 a, int256 b) internal pure returns (int256) {
+        return a.add(b);
+    }
+
+    /**
+     * @notice return the difference of two signed integer
+     * @param a signed integer
+     * @param b signed integer
+     * @return difference of two signed integer
+     */
+    function sub(int256 a, int256 b) internal pure returns (int256) {
+        return a.sub(b);
+    }
+
+    /**
+     * @notice multiply two signed integer
+     * @dev rounds to zero if a*b < SCALING_FACTOR / 2
+     * @param a signed integer
+     * @param b signed integer
+     * @return mul of two signed integer
+     */
+    function mul(int256 a, int256 b) internal pure returns (int256) {
+        return (a.mul(b)).add(SCALING_FACTOR / 2) / SCALING_FACTOR;
```

Comment:
```suggestion
        return (a.mul(b)).add(SCALING_FACTOR / 2).div(SCALING_FACTOR);
```

Any reason why not using safemath `.div` here?

Reply:
It is the same, `SCALING_FACTOR` is always equal to 1e18.